java/src/jmri/jmrix/logixng/ActionRequestUpdateAllSensors.java

```diff
@@ -0,0 +1,119 @@
+package jmri.jmrix.logixng;
+
+import jmri.jmrix.loconet.logixng.*;
+import jmri.jmrit.logixng.actions.*;
+
+import java.util.Locale;
+import java.util.Map;
+
+import jmri.*;
+import jmri.jmrit.logixng.*;
+
+/**
+ * Sets all engine slots to status common
+ *
+ * @author Daniel Bergqvist Copyright 2022
+ */
+public class ActionRequestUpdateAllSensors extends AbstractDigitalAction {
+
+    private SystemConnectionMemo _memo;
+
+    public ActionRequestUpdateAllSensors(String sys, String user, SystemConnectionMemo memo) {
+        super(sys, user);
+        _memo = memo;
+    }
+
+    @Override
+    public Base getDeepCopy(Map<String, String> systemNames, Map<String, String> userNames) throws JmriException {
+        DigitalActionManager manager = InstanceManager.getDefault(DigitalActionManager.class);
+        String sysName = systemNames.get(getSystemName());
+        String userName = userNames.get(getSystemName());
+        if (sysName == null) sysName = manager.getAutoSystemName();
+        ActionRequestUpdateAllSensors copy = new ActionRequestUpdateAllSensors(sysName, userName, _memo);
+        return manager.registerAction(copy).deepCopyChildren(this, systemNames, userNames);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Category getCategory() {
+        return Category.ITEM;
+    }
+
+    public void setMemo(SystemConnectionMemo memo) {
+        assertListenersAreNotRegistered(log, "setMemo");
+        _memo = memo;
+    }
+
+    public SystemConnectionMemo getMemo() {
+        return _memo;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void execute() {
+        if (_memo != null) {
+            if (_memo.provides(SensorManager.class)) {
+//                _memo.get(SensorManager.class).updateAll();
+                ((SensorManager)_memo.get(SensorManager.class)).updateAll();
```

Comment:
@bobjacobsen 
Am I missing something here? I expected `_memo.get(SensorManager.class).updateAll();` to work but that gives a compile error. But `((SensorManager)_memo.get(SensorManager.class)).updateAll();` works fine.

I thought that the compiler would see that  `_memo.get(SensorManager.class)` returns a SensorManager object, but it doesn't seem to do that.